### ISSUE TITLE
Unable to interpret notation as a reference

### ISSUE THREAD

**Issue by samuelgruetter** [coq/coq#11420]
```coq
Require Import Coq.NArith.NArith. Open Scope N_scope.

Goal forall (a: N), a - a = 0.
Proof.
  unfold "-". (* succeeds *)
Abort.

Goal forall (a: N), (a | a).
Proof.
  unfold "|". (* Error: Unable to interpret "|" as a reference. *)
```

In the above example, I can unfold `-`, but if I try to do the same for `|`, it fails.

Desired behavior: `unfold "|"` should do the same as `unfold N.divide`.

Workaround: I will try rotating my screen by 90° :wink:

Coq Version: 8.9 and master (37254871c8e5ece576af7efddc20a9ed7f197e04)

**Comment by ejgallego** [comment]
Yeah that's little known, but you need to use the notation full form in this case `unfold "( _ | _ )".`

**Comment by ejgallego** [comment]
I'm tempted to say this is hard to solve, "|" is too ambiguous as to resolve it properly in your context.

**Comment by samuelgruetter** [comment]
`Locate "|".` indeed lists quite a few notations, so I can understand why you call this "too ambiguous". I did try the full form, but I wrote `unfold "(_ | _)"`, which does not work, only the version with additional spaces, `unfold "( _ | _ )"` works. Why is this?
